.github/workflows/pull-request.yml

```diff
@@ -83,10 +101,22 @@ jobs:
           ~/.gradle/wrapper
         key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
         restore-keys: ${{ runner.os }}-gradle
+    - name: Build/test options
+      id: ci_options
+      run: |
+        echo ::set-output name=CI_WITH_NATIVE::$(jq 'any(.pull_request.labels[]; .name == "pr-native")' < $GITHUB_EVENT_PATH)
+        echo ::set-output name=CI_WITH_JDK8::$(jq 'any(.pull_request.labels[]; .name == "pr-jdk8")' < $GITHUB_EVENT_PATH)
+        echo ::set-output name=CI_WITH_JACKSON::$(jq 'any(.pull_request.labels[]; .name == "pr-jackson")' < $GITHUB_EVENT_PATH)
     - name: Build with Maven
-      run: ./mvnw -B install javadoc:javadoc-no-fork --file pom.xml -Pcode-coverage,jdk8-tests -Dtest.log.level=WARN
       env:
+        CI_WITH_NATIVE: ${{ steps.ci_options.outputs.CI_WITH_NATIVE }}
+        CI_WITH_JDK8: ${{ steps.ci_options.outputs.CI_WITH_JDK8 }}
         SPARK_LOCAL_IP: localhost
+      run: |
+        PROFILES="-Pcode-coverage"
+        [[ $CI_WITH_JDK8 == "true" ]] && PROFILES="$PROFILES,jdk8-tests"
```

Comment:
can these be split up into different runs? They would make the build easier to read and we would see that jobs are actually run. It might prevent having to re-run the whole build for a single label change. I would rather be able to view when they have run and their results/logs. Especially for new users.

Reply:
Moved those to separate `run`s